src/entities/relay.ts

```diff
@@ -0,0 +1,70 @@
+import { type DocumentLoader, Follow, Undo } from "@fedify/fedify";
+
+// Use fixed username and id for the relay client actor
+// Why use fixed username and id?
+// Because the relay client actor is a special actor that is not supposed to be interacted with by users.
+// Also, if we delete the a relay and send the Undo activity, the relay client needs to be online.
+// For example, AodeRelay first fetches the actor from our server again, and if we already deleted the actor, it will fail
+// and the realy keeps our server in the database.
+// And because we won't receive an activity to check whether the Undo was successful, we can't delete the relay client actor safely.
+export const HOLLO_RELAY_ACTOR_ID = "8a683714-6fa2-4e53-9f05-b4acbcda4db7";
+export const HOLLO_RELAY_ACTOR_USERNAME = "hollo-relay-follower";
+
+/**
+ * Workaround fedify jsonld serialization for relay follow.
+ *
+ * Without this, the jsonld serialization would be:
+ * ```json
+ * {
+ *  "@context": "https://www.w3.org/ns/activitystreams",
+ *  "id": "https://client.example/6ae15297",
+ *  "type": "Follow",
+ *  "actor": "https://client.example/actor",
+ *  "object": "as:public"
+ * }
+ * ```
+ * instead of
+ * ```json
+ * {
+ *  "@context": "https://www.w3.org/ns/activitystreams",
+ *  "id": "https://client.example/6ae15297",
+ *  "type": "Follow",
+ *  "actor": "https://client.example/actor",
+ *  "object": "https://www.w3.org/ns/activitystreams#Public"
```

Comment:
He the relay should actually understand both, as they're semantically the same, I'd file a bug with the relay implementation

Reply:
I am not sure about this.

> The object of the Follow request MUST be the fully expanded URI of the Public pseudo-collection (https://www.w3.org/ns/activitystreams#Public).

As far as I understand, but you can correct me:

- `https://www.w3.org/ns/activitystreams#Public` fully expanded
- `as:public` not fully expanded

Imagine building a new relay software: I would look at the spec and see that the object property should be equal to `https://www.w3.org/ns/activitystreams#Public`. So I would just check for this. This is why I think using `https://www.w3.org/ns/activitystreams#Public` would make it more compatible with more relay softwares. But at this point I only tested one software.